### DISCUSSION TITLE
Is there any way to get the color and direction of rotation of an annotation?

### DISCUSSION THREAD

**Discussion by izerui** [pymupdf/PyMuPDF#3333] [Q&A] [answered]
### Description of the bug

I expect to get a new page with just the annotation

like this:
![22](https://github.com/pymupdf/PyMuPDF/assets/1773233/2047d46c-624f-429a-ac29-1e90503716d6)

### How to reproduce the bug

When I try to get the raw attributes of an annotation, like color, rotation.

```       
    def test_copy_annot(self):
        bytes = get_url_content_retry('https://tfile.yj2025.com/pdf-processor/source/mt_04_24024_0_812.pdf')
        target_doc: Document = fitz.open()
        doc = fitz.open('pdf', bytes)
        for index, page in enumerate(doc):
            new_page = target_doc.new_page(width=page.rect.width, height=page.rect.height)
            for index, annot in enumerate(page.annots()):
                print(annot.type, annot.info['content'], annot.colors["stroke"], annot.rotation)
            pass
        target_doc.save('111.pdf', garbage=4, deflate=True)
        doc.close()
```

But the output rotation is always  [1.0, 1.0, 1.0]  and  -1.

output :
```
(2, 'FreeText') MT_04_24024_0_812 [1.0, 1.0, 1.0] -1
(2, 'FreeText')    MT_04_24024_0_812 [1.0, 1.0, 1.0] -1
(2, 'FreeText') MS          CZ        CW [1.0, 1.0, 1.0] -1
(2, 'FreeText') 2024-04-01 [1.0, 1.0, 1.0] -1
(2, 'FreeText') 数量：2（双台） [1.0, 1.0, 1.0] -1
(2, 'FreeText') *41.455士0.1 [1.0, 1.0, 1.0] -1
(2, 'FreeText') 20.728 [1.0, 1.0, 1.0] -1
```

and result pic is:
![result](https://github.com/pymupdf/PyMuPDF/assets/1773233/b401d260-98dc-4542-80ae-abedc48814f0)

I did this because I merged two pages using `page.show_pdf_page`, but the annotations on one of the pages could not be copied over.

### PyMuPDF version

1.24.0

### Operating system

MacOS

### Python version

3.10

**Comment by JorjMcKie** [comment]
This is not a bug, but a question. Converting to a Discussions items.

**Reply by izerui** [reply]
@JorjMcKie yes, i got it!!

Yes, how to get the text_color, font_size, rotation of annotation through pymupdf. 
Or is there any way to copy the annotation information. Because show_pdf_page can't copy them.

**Reply by jamie-lemon** [reply]
Just appreciating that I think these might actually be rich text attributes.

**Reply by JorjMcKie** [reply]
(Py-)MuPDF does not support rich text in annotations at all.

**Reply by izerui** [reply]
Yes, but find another way to get some of the properties in the rich text, such as passing

```
print('Rotation:', doc.xref_get_key(annot.xref, 'Rotation'))
                print('Contents:', doc.xref_get_key(annot.xref, 'Contents'))
                print('Default Style:', doc.xref_get_key(annot.xref, 'DS'))
                print('Default Appearance:', doc.xref_get_key(annot.xref, 'DA'))
                print('Remote Control:', doc.xref_get_key(annot.xref, 'RC'))
```

@_@

**Reply by izerui** [reply]
I'm just puzzled that the annot object of type FreeText has rotation and color in its properties, but these values are incorrect.

**Reply by JorjMcKie** [reply, marked as answer]
No - here is a misconception:
PDF only supports 2 different colors in all of its object definitions (i.e. including annotations): Stroke color and fill color. So, by design of PDF, there is no way to specify (or inquire) all 3 desired colors: border, fill and text color.

The text color specifically must be inquired by extracting the text in the normal way.

You could of course extract the `/RC`, `/DS` contents (XML/HTML formatted strings). Then use some package to parse them. This is beyond PyMuPDF`s scope.
